### ISSUE TITLE
Audit example READMEs against NEAT-AI COMPARISON.md feature list

### ISSUE THREAD

**Issue by stservice** [stSoftwareAU/NEAT-AI-Examples#184]
## Summary
Produce an audit document that cross-references every README in this repository against the canonical [`NEAT-AI/COMPARISON.md`](https://github.com/stSoftwareAU/NEAT-AI/blob/Develop/COMPARISON.md) feature list, listing inaccuracies and gaps where the docs imply NEAT-AI is "just textbook NEAT" rather than the broader toolkit it actually is.

## What Needs to Be Done
- Add a new file `docs/neat_ai_feature_audit.md` that:
  - Lists every NEAT-AI capability called out in the upstream `COMPARISON.md` (backpropagation, memetic evolution, MCMC mutation acceptance, GPU-accelerated discovery, predictive coding, dropout, L1/L2 regularisation, K-fold cross-validation, synthetic synapse training, advanced breeding, hyperparameter self-adaptation, adaptive population sizing, ONNX export, transfer learning, etc.).
  - For each capability, records (a) which READMEs already mention it, (b) which READMEs misrepresent it (e.g. claim NEAT lacks it), and (c) which examples in this repo exercise it.
  - Flags every passage that uses the unqualified word "NEAT" where "standard/textbook NEAT" is actually meant (so it does not read as a limitation of NEAT-AI).
  - Captures the user's specific points from issue #182: the "back propagation" framing in `mnist_classification/README.md`, the binary-data-format speed advantage, and NEAT-AI-Discovery's "science-driven" structural mutation.
- Include a Mermaid summary diagram that groups the features by category (search, training, structure, scale, interop).
- The audit is the source of truth used by the follow-up sub-issues; it does not itself rewrite any README.

## Acceptance Criteria
- [ ] `docs/neat_ai_feature_audit.md` exists and is referenced from the parent `README.md` "Related Repositories" section so future contributors find it.
- [ ] Every README under `*/README.md` is listed in the audit table, with at least one row noting current accuracy (or "no NEAT-vs-NEAT-AI claim made").
- [ ] Each NEAT-AI feature from upstream `COMPARISON.md` appears in the audit and is tagged either "demonstrated by example X" or "not yet demonstrated".
- [ ] A unit test (`docs/neat_ai_feature_audit_test.ts`) verifies the audit file exists, contains a non-empty Mermaid block, and references every per-example README path that currently lives in the repo.
- [ ] `./quality.sh` passes.

## Dependencies
None — this is the foundation.

## Diagram
```mermaid
flowchart LR
    AUDIT[#NNN Documentation audit] --> MNIST[MNIST README rewrite]
    AUDIT --> TOP[Top-level README breadth section]
    AUDIT --> CALLOUT[Per-example feature callouts]
    AUDIT --> SYN[Synthetic Synapse clarification]
    AUDIT --> DISC[Discovery 'applies science' framing]
```

## Context
Part of #182

The reporter notes that `mnist_classification/README.md` reads as if NEAT-AI lacks backpropagation, when in fact backprop is one of many NEAT-AI training features (see upstream `COMPARISON.md`). This audit ensures the follow-up README rewrites cover every feature consistently, instead of fixing one passage at a time.

**Comment by stservice** [comment]
Pull request https://github.com/stSoftwareAU/NEAT-AI-Examples/pull/192 has been created to address this issue.
